trackintel/analysis/location_identification.py

```diff
@@ -0,0 +1,229 @@
+import numpy as np
+import pandas as pd
+import trackintel as ti
+
+
+def location_identifier(sps, recipe="FREQ", pre_filter=True, **pre_filter_kwargs):
+    """Assign "home" and "work" activity label for each user with different recipes.
+
+    Parameters
+    ----------
+    sps : Geodataframe (as trackintel staypoints)
+        Staypoints with column "location_id".
+
+    recipe : {'FREQ'}, default "FREQ"
+        Choose which recipe to use.
+        - FREQ: "Generate a activity label per user by assigning the most visited location the label "home"
+          and the second most visited location the label "work". The remaining locations get no label.
+
+    pre_filter : bool, default True
+        Prefiltering the staypoints to exclude locations with not enough data.
+        The filter function can also be accessed via `pre_filter_locations`.
+
+    pre_filter_kwargs : dict
+        Kwargs to hand to `pre_filter_locations` if used. See function for more informations.
+
+    Returns
+    -------
+    Geodataframe (as trackintel staypoints)
+        With additional column `activity label` assigning one of three activity labels {'home', 'work', None}.
+
+    Note
+    ----
+    The recipe are adapted from [1]. The original algorithms count the distinct hours at a
+    location as the home location is derived from geo-tagged tweets. We directly sum the time
+    spent at a location as our data model includes that.
+
+    References
+    ----------
+    Chen, Qingqing, and Ate Poorthuis. 2021.
+    ‘Identifying Home Locations in Human Mobility Data: An Open-Source R Package for Comparison and Reproducibility’.
+    International Journal of Geographical Information Science 0 (0): 1–24.
+    https://doi.org/10.1080/13658816.2021.1887489.
+
+    Examples
+    --------
+    >>> ti.analysis.location_identifier(sps, pre_filter=True, recipe="FREQ")
+    """
+    assert sps.as_staypoints
+    sps.copy()
```

Comment:
I think you need to write `sps = sps.copy()`
`.copy()` is not an inplace operation